### PR TITLE
[Merged by Bors] - feat: GCDMonoid (Associates α) instance

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/GCDMonoid/Basic.lean [hunk 1]
```diff
@@ -1420,3 +1420,42 @@ theorem normalize_eq_one {a : G₀} (h0 : a ≠ 0) : normalize a = 1 := by simp
 #align comm_group_with_zero.normalize_eq_one CommGroupWithZero.normalize_eq_one
 
 end CommGroupWithZero
+
+namespace Associates
+
+noncomputable instance [CancelCommMonoidWithZero α] [GCDMonoid α] : GCDMonoid (Associates α) where
+  gcd a b := Associates.mk (gcd (Quot.out a) (Quot.out b))
+  lcm a b := Associates.mk (lcm (Quot.out a) (Quot.out b))
```
Comment (eric-wieser) [inline Mathlib/Algebra/GCDMonoid/Basic.lean:1428]: Is it true that `lcm (Associates.mk a) (Associates.mk b) = Associates.mk (lcm a b)`? If so can you prove it; and if not, are you sure this instance is a good idea?

Comment (eric-wieser) [reply Mathlib/Algebra/GCDMonoid/Basic.lean:1428]: Which is to say; is the following sorry true?
```lean
theorem Associated.gcd [CancelCommMonoidWithZero α] [GCDMonoid α]
    {a₁ a₂ b₁ b₂ : α} (ha : Associated a₁ a₂) (hb : Associated b₁ b₂) :
    Associated (gcd a₁ b₁) (gcd a₂ b₂) :=
  sorry

noncomputable instance [CancelCommMonoidWithZero α] [GCDMonoid α] : GCDMonoid (Associates α) where
  gcd := Quotient.map₂' gcd fun a₁ a₂ (ha : Associated _ _) b₁ b₂ (hb : Associated _ _) => ha.gcd hb
```

Comment (eric-wieser) [reply Mathlib/Algebra/GCDMonoid/Basic.lean]: It is, pushed :). Please add the lemma I ask for above, which is now true by `rfl`.

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**